lms/djangoapps/program_enrollments/management/commands/send_program_course_nudge_email.py

```diff
@@ -79,19 +81,67 @@ def valid_course_run(self, course_run):
             and course_run['image'] \
             and course_run['status'] == COURSE_PUBLISHED
 
-    def get_course_run_to_suggest(self, programs_progress, completed_course_id):
+    @staticmethod
+    def get_candidate_program_and_courses(program, program_progress):
+        """
+        Get program in a new format containing candidate_courses and in_progress_courses_ids
+        """
+        if program['uuid'] != program_progress['uuid']:
+            raise RuntimeError('Program and Program Progress should be same')
+        not_started_courses_ids = [course['key'] for course in program_progress['not_started']]
+        in_progress_courses_ids = [course['key'] for course in program_progress['in_progress']]
+        candidate_courses = not_started_courses_ids + in_progress_courses_ids
+        program_copy = deepcopy(program)
```

Comment:
Why do we need to create a deep copy here?

Reply:
We are pointing courses of this program in a new dictionary. Making a deep copy here to make sure any change in the new dictionary cannot affect the actual program object.